spotbugs/src/main/java/edu/umd/cs/findbugs/BugPattern.java

```diff
@@ -36,6 +37,8 @@
  * @see BugInstance
  */
 public class BugPattern implements Comparable<BugPattern> {
+    final private String ruleId;
```

Comment:
Consider to add a documentation comment, especially a `@see` tag with an anchor tag to http://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html#def_rule_id